src/ui/component/dayPicker/dayPicker.tsx

```diff
@@ -0,0 +1,60 @@
+import { type FC, useMemo } from 'react'
+import type { SelectSingleEventHandler } from 'react-day-picker'
+import { DayPicker as RDayPicker } from 'react-day-picker'
+import { enUS, fr, de } from 'date-fns/locale'
+import i18n from '@/ui/i18n'
+import 'react-day-picker/dist/style.css'
+import './dayPicker.scss'
+
+const localeMap = new Map([
+  ['en', enUS],
+  ['fr', fr],
+  ['de', de]
+])
+
+type DayPickerProps = {
+  onSelect: SelectSingleEventHandler
```

Comment:
Avoid exposing the type of the handler. Additionally, if this type changes, it may break the interface of the component.

Reply:
The function signature provided by the library is the following:
```ts
(day?: Date, selectedDay: Date, activeModifiers: ActiveModifiers, e: React.MouseEvent) => void;
```
We can provide the detailed handler signature for readability and maintainability.

Reply:
The function signature provided by the library is the following:
```ts
(day?: Date, selectedDay: Date, activeModifiers: ActiveModifiers, e: React.MouseEvent) => void;
```
We can provide the detailed handler signature for readability and maintainability.

Reply:
Yes. Up to you to choose what to expose, you're in control 🚗.  The idea is to design our own types rather than directly exposing types from third-party libraries. :relieved: